### PR TITLE
Feature to add logical roles

### PR DESCRIPTION
**Pull request by TiBarification**
More complex system of role check.

More details from #175

### CODE REVIEW

File: SCPDiscordPlugin/LogicRole.cs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+		public LogicType Type { get; set; }
+		public List<ulong> Roles { get; set; }
+		[CanBeNull] public List<string> Commands { get; set; }
+		[CanBeNull] public Dictionary<int, LogicRole> Children { get; set; }
+
+		public bool IsPermitted(List<ulong> userRoles)
+		{
+			return Type switch
+			{
+				LogicType.None => true,
```
Comment (TiBarification) [inline SCPDiscordPlugin/LogicRole.cs]: Oops, here should be false. Means we never go execute commands and go into childrens

**Review by KarlOfDuty — CHANGES_REQUESTED** [review]
This system is way too complicated, you need to either simplify it or change the config to be much easier to understand.

**Comment by TiBarification** [comment]
Its purpose is not to simplify, but to make it more flexible in management. I think I will close pr then because it doesn't correlate with your view.

**Comment by KarlOfDuty** [comment]
I understand that, but most people only want to sync specific roles to the game, not combinations of roles. The system you have created makes it much more complicated to configure for everyone, and only a small number of people will need it.

**Pull request closed without merging**